src/newcharacter.cpp

```diff
@@ -535,32 +535,13 @@ bool player::create(character_type type, std::string tempname)
         }
     }
 
-    std::vector<item> prof_items = g->u.prof->items( g->u.male );
-
-    // Those with certain special traits are guaranteed to start with certain items
-    if( has_trait( "HYPEROPIC" ) && has_trait( "MYOPIC" ) ) {
-        prof_items.push_back( item( "glasses_bifocal" ) );
-    }
-    else if( has_trait( "MYOPIC" ) ) {
-        prof_items.push_back( item( "glasses_eye" ) );
-    }
-    else if( has_trait( "HYPEROPIC" ) ) {
-        prof_items.push_back( item( "glasses_reading" ) );
-    }
-
-    if( has_trait( "ASTHMA" ) ) {
-        prof_items.push_back( item( "inhaler", 0, item::default_charges_tag{} ) );
-    }
-    if (has_trait("CANNIBAL")) {
-        prof_items.push_back( item( "cookbook_human", 0 ) );
-    }
-    if( has_trait( "ALBINO" ) ) {
-        prof_items.push_back( item( "teleumbrella", 0 ) );
-    }
+    std::list<item> prof_items = g->u.prof->items( g->u.male, g->u.get_mutations() );
 
     for( item &it : prof_items ) {
         if( it.is_armor() ) {
-            wear_item( it, false ); // If wearing fails, we fail silently
```

Comment:
I plan to make the `json_item_substitution`s cover every case of wardrobe (and food) failure, and the point of that change is to note any `json_item_substitution`s that weren't added, indicating an accidental deficiency in the test case (e.g. I forgot about hooves, and the test doesn't cover any of the cases DangerNoodle mentioned).

Unless certain cases are intentionally unaccounted for by the `json_item_substitution`s, it should be treated as an error. So the question becomes, why not cover every scenario?